schema/pgsql/upgrades/024.sql

```diff
@@ -0,0 +1,3 @@
+ALTER TABLE incident_history
+    ALTER COLUMN event_id SET NOT NULL,
```

Comment:
Didn't verify this myself if it's actually the case, but wouldn't that cause problems with existing setups and have `event_id` set to NULL in the history?

Reply:
Well.. this might be the case. But which existing setups?

Reply:
> Well.. this might be the case. But which existing setups?

Existing setups of our web colleges 😄! I don't think they will be willingly wiping out their history entries, but if @nilmerg says otherwise, then I don't mind.

Reply:
But what would be a sane fallback as it `REFERENCES event(id)`. I could create a placeholder `event` with its placeholder `object` on the first available `source`, but this would "corrupt" the database even more, IMO.

Thus, I would prefer the schema upgrade to fail and let the users (our web colleges) handle with it.

Reply:
This isn't just an issue during migration of existing setups. I missed that Web creates history entries on its own, when changing recipient roles. In that case, it would also need to emit its own internal events? Eh!

Reply:
@nilmerg: I added the `NOT NULL` clause on your request in #163. Thus, should I remove it again?

Reply:
I know. :wink: I wanted to say that this needs to be solved first, before finishing this. The `event_id` being non nullable is and should be fine. 

To answer Yonas' argument: Web could handle this on its own, though a `DELETE FROM incident_history where event_id IS NULL` in the upgrade file wouldn't hurt either.

Reply:
K. I have added the `DELETE` into the schema upgrade file.

Reply:
So if I understand correctly, merging this as-is would break the (un)manage/subscribe buttons in Web due to this `not null` constraint, wouldn't it?

@nilmerg Should this wait for something in Web therefor?

I think there will be another PR anyways which will add a new type for Web to use in the `event_type` enum introduced in #175, so adding the constraint could also be postponed to that PR until Web is ready for it?

Reply:
+1 for postponing the constraint change